### PR TITLE
cilium/cmd: cilium status --verbose and cilium-health status --succinct support show IPv6 IP

### LINKED ISSUE #27828: cilium status --verbose doesn't show IPv6 IP
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

The `cilium status --verbose` command doesn't show the IPv6 address

```
Cluster health:             1/3 reachable   (2023-08-29T15:34:44Z)
  Name                      IP              Node          Endpoints
  redacted (localhost)      redacted        reachable     reachable
  redacted                  redacted        unreachable   reachable
  redacted                  redacted        unreachable   reachable
```

Which is confusing when double-checking with `cilium-health status --verbose`

```
Probe time:   2023-08-29T15:34:44Z
Nodes:
  redacted (localhost):
    Host connectivity to redacted:
      ICMP to stack:   OK, RTT=393.615µs
      HTTP to agent:   OK, RTT=362.15µs
      Secondary connectivity to redacted
        ICMP to stack:   OK, RTT=440.104µs
        HTTP to agent:   OK, RTT=252.664µs
    Endpoint connectivity to redacted:
      ICMP to stack:   OK, RTT=420.002µs
      HTTP to agent:   OK, RTT=321.53µs
      Secondary connectivity to redacted:
        ICMP to stack:   OK, RTT=400.24µs
        HTTP to agent:   OK, RTT=278.436µs
  toroon63okm:
    Host connectivity to redacted:
      ICMP to stack:   OK, RTT=423.775µs
      HTTP to agent:   OK, RTT=651.82µs
      Secondary connectivity to redacted:
        ICMP to stack:   Connection timed out
        HTTP to agent:   Get "http://[redacted]:4240/hello": context deadline exceeded (Client.Timeout exceeded while awaiting 
```

### Cilium Version

1.11.8

### CLI Version

v0.15.5

### Kernel Version

4.18.0-305.57.1.el8_4.x86_64

### Kubernetes Version

v1.23.5

### Sysdump

_No response_

### Relevant log output

_No response_

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by chaunceyjiang**
Signed-off-by: chaunceyjiang <chaunceyjiang@gmail.com>

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

Fixes: #27828

Test:

image

image

```release-note
Improve `cilium status --verbose` and `cilium-health status --succinct` support to show IPv6 IPs as well
```

### CODE REVIEW

**Comment by chaunceyjiang** [comment]
Test:

image


image

**Comment by chaunceyjiang** [comment]
/cc @ldelossa 

Ready for review. Please take a look. Thanks~

File: pkg/health/client/client.go [hunk 1]
```diff
@@ -317,8 +317,18 @@ func formatNodeStatus(w io.Writer, node *models.NodeStatus, printAll, succinct,
...
+				if addr == nil {
+					ip = "nil"
+				} else {
+					ip = addr.IP
+				}
+				ips = append(ips, ip)
+			}
 			fmt.Fprintf(w, "  %s%s\t%s\t%s\t%s\n", node.Name,
-				localStr, getPrimaryAddressIP(node),
+				localStr, strings.Join(ips, ","),
```
Comment (christarazi) [inline pkg/health/client/client.go]: If any address is somehow nil, then could the list show something like

```
1.1.1.1,nil,::1         reachable
```
?

If so, I think that's undesirable.

Comment (chaunceyjiang) [reply pkg/health/client/client.go]: Updated.

**Review by christarazi — APPROVED** [review]

**Comment by christarazi** [comment]
BPF checkpatch is failing with:

```
Error: ERROR:CUSTOM: Please avoid long commit subjects (max: 75, found: 80)
```

so it looks like you'll need to shorten the commit title.

**Comment by chaunceyjiang** [comment]
> so it looks like you'll need to shorten the commit title.



commit message: "cilium status and cilium-health status support show IPv6 IP"

Updated. /cc @christarazi  PATL. Thank~

**Comment by chaunceyjiang** [comment]
It seems GitHub is having troubles. Let me close and re-open the PR, that should re-trigger the workflows.

**Comment by chaunceyjiang** [comment]
Hi, @christarazi Can you help me run E2E tests again?

**Comment by chaunceyjiang** [comment]
It seems GitHub is having troubles.

**Comment by christarazi** [comment]
The non-required tests failed, so the CI is effectively passing for this PR. Just waiting on @ldelossa's review.

**Review by ldelossa — APPROVED** [review]
LGTM

**Pull request merged**